code/game/machinery/shieldgen.dm

```diff
@@ -553,3 +553,644 @@
+	var/active = FALSE
+
+	///Determins if we can turn it on or not, no longer recovering when back to max strength
+	var/recovering = TRUE
+
+	///Determins max health of the shield
+	var/max_strength = 40
+
+	///Current health of shield
+	var/stored_strength = 0 //starts at 0 to prevent rebuild abuse
+
+	///Shield Regeneration when at 100% efficiency
+	var/max_regeneration = 3
+
+	///The regeneration that the shield can support
+	var/current_regeneration
+
+	///Determins the max radius the shield can support
+	var/max_radius = 3
+
+	///Current radius the shield is set to, minimum 3
+	var/radius = 3
+
+	///Determins if we only generate a shield on space turfs or not
+	var/exterior_only = FALSE
+
+	///The list of shields that are ours
+	var/list/deployed_shields = null
+
+	///The list of turfs that are within the shield
+	var/list/inside_shield = null
+
+	///The list of machines that are connected to and boosting us
+	var/list/obj/machinery/modular_shield/module/connected_modules = null
+
+	///Regeneration gained from machines connected to us
+	var/regen_boost = 0
+
+	///Max Radius gained from machines connected to us
+	var/radius_boost = 0
+
+	///Max Strength gained from machines connected to us
+	var/max_strength_boost = 0
+
+	///Regeneration gained from our own parts
+	var/innate_regen
+
+	///Max radius gained from our own parts
+	var/innate_radius
+
+	///Max strength gained from our own parts
+	var/innate_strength
+
+	var/list/list_of_turfs
+
+/obj/machinery/modular_shield_gen/power_change()
+	. = ..()
+	if(machine_stat & NOPOWER)
+		deactivate_shields()
+		STOP_PROCESSING(SSobj, src)
+		return
+	START_PROCESSING(SSobj, src)
+
+/obj/machinery/modular_shield_gen/RefreshParts()
+	. = ..()
+
+	innate_regen = 3
+	innate_radius = 3
+	innate_strength = 40
+
+	for(var/datum/stock_part/capacitor/new_capacitor in component_parts)
+		innate_strength += new_capacitor.tier * 10
+
+	for(var/datum/stock_part/servo/new_servo in component_parts)
+		innate_regen += new_servo.tier
+
+	for(var/datum/stock_part/micro_laser/new_laser in component_parts)
+		innate_radius += new_laser.tier * 0.25
+
+	calculate_regeneration()
+	calculate_max_strength()
+	calculate_radius()
+
+
+/obj/machinery/modular_shield_gen/Initialize(mapload)
+	. = ..()
+	deployed_shields = list()
+	connected_modules = list()
+	list_of_turfs = list()
+	wires = new /datum/wires/modular_shield_gen(src)
+	if(mapload && active && anchored)
+		activate_shields()
+
+/datum/wires/modular_shield_gen
+	proper_name = "Modular shield generator"
+	randomize = FALSE
+	holder_type = /obj/machinery/modular_shield_gen
+
+/datum/wires/modular_shield_gen/New(atom/holder)
+	wires = list(WIRE_HACK)
+	..()
+
+/datum/wires/modular_shield_gen/on_pulse(wire)
+
+	var/obj/machinery/modular_shield_gen/G = holder
+	switch(wire)
+		if(WIRE_HACK)
+			G.toggle_shields()
+			return
+	..()
+
+/obj/machinery/modular_shield_gen/proc/deactivate_shields()
```

Comment:
you should be DMdoccing all the new procs you're adding, if for no other reason but to help statistics. https://github.com/tgstation/tgstation/blob/master/.github/guides/AUTODOC.md